addons/website_sale/controllers/main.py

```diff
@@ -89,55 +87,49 @@ def __init__(self, path='', **args):
     def __call__(self, path=None, **kw):
         if not path:
             path = self.path
-        for k,v in self.args.items():
-            kw.setdefault(k,v)
+        for k, v in self.args.items():
+            kw.setdefault(k, v)
         l = []
-        for k,v in kw.items():
+        for k, v in kw.items():
             if v:
                 if isinstance(v, list) or isinstance(v, set):
-                    l.append(werkzeug.url_encode([(k,i) for i in v]))
+                    l.append(werkzeug.url_encode([(k, i) for i in v]))
                 else:
-                    l.append(werkzeug.url_encode([(k,v)]))
+                    l.append(werkzeug.url_encode([(k, v)]))
         if l:
             path += '?' + '&'.join(l)
         return path
 
-
 def get_pricelist():
     return request.website.get_current_pricelist()
 
-class website_sale(http.Controller):
+
+class WebsiteSale(http.Controller):
 
     def get_pricelist(self):
         return get_pricelist()
 
     def get_attribute_value_ids(self, product):
-        cr, uid, context, pool = request.cr, request.uid, request.context, request.registry
-        currency_obj = pool['res.currency']
         attribute_value_ids = []
-        visible_attrs = set(l.attribute_id.id
-                                for l in product.attribute_line_ids
-                                    if len(l.value_ids) > 1)
-        if request.website.pricelist_id.id != context['pricelist']:
-            website_currency_id = request.website.currency_id.id
-            currency_id = self.get_pricelist().currency_id.id
+        visible_attrs_ids = product.mapped('attribute_line_ids.attribute_id').filtered(lambda attr: len(attr.value_ids) > 1).ids
+        if request.website.pricelist_id.id != request.env.context.get('pricelist'):
+            to_currency = self.get_pricelist().currency_id
             for p in product.product_variant_ids:
-                price = currency_obj.compute(cr, uid, website_currency_id, currency_id, p.lst_price)
-                attribute_value_ids.append([p.id, [v.id for v in p.attribute_value_ids if v.attribute_id.id in visible_attrs], p.price, price])
+                price = request.website.currency_id.compute(p.lst_price, to_currency)
+                attribute_value_ids.append([p.id, p.attribute_value_ids.filtered(lambda val: val.attribute_id.id in visible_attrs_ids).ids, p.price, price])
         else:
-            attribute_value_ids = [[p.id, [v.id for v in p.attribute_value_ids if v.attribute_id.id in visible_attrs], p.price, p.lst_price]
+            attribute_value_ids = [[p.id, p.attribute_value_ids.filtered(lambda val: val.attribute_id.id in visible_attrs_ids).ids, p.price, p.lst_price]
```

Comment:
The advantage of using `filtered` is that you are working on a recordset instead of a list. In this case you just make `.ids` at the end so ends up with a list. It makes the line longer with badly named variable (`val`) so reduce the readability of the list. Use `mapped` and `filtered` when it makes sense to do it, do not blindly convert all.